### PR TITLE
DOC: update the pandas.date_range() docstring

### PR DESCRIPTION
**Pull request by nipunsadvilkar**
Checklist for the pandas documentation sprint (ignore this if you are doing
an unrelated PR):

- [x] PR title is "DOC: update the <your-function-or-method> docstring"
- [x] The validation script passes: `scripts/validate_docstrings.py <your-function-or-method>`
- [x] The PEP8 style check passes: `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] The html version looks good: `python doc/make.py --single <your-function-or-method>`
- [x] It has been proofread on language by another sprint participant

Please include the output of the validation script below between the "```" ticks:

```
Errors found:
	Errors in parameters section
		Parameters {'kwargs'} not documented
```

If the validation script still gives errors, but you think there is a good reason
to deviate in this case (and there are certainly such cases), please state this
explicitly.

Checklist for other PRs (remove this part if you are doing a PR for the pandas documentation sprint):

- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by jazzmuesli** [comment]
Can you insert the output of scripts/validate_docstrings.py?

**Comment by nipunsadvilkar** [comment]
Just did

File: pandas/core/indexes/datetimes.py [hunk 1]
```diff
@@ -2237,6 +2238,22 @@ def date_range(start=None, end=None, periods=None, freq='D', tz=None,
...
     rng : DatetimeIndex
+
+    See Also
+    ________
+    pandas.period_range : Return a fixed frequency PeriodIndex.
+    pandas.interval_range : Return a fixed frequency IntervalIndex.
+
+    Examples
+    --------
+    >>> import pandas
```
Comment (TomAugspurger) [inline pandas/core/indexes/datetimes.py]: We assume that these imports are available

```python
import pandas as pd
import numpy as np
```

so you can remove the import and do `pd.range_range`

**Comment by jazzmuesli** [comment]
Looks like the validation fails?
Errors found:
	Errors in parameters section
		Parameters {'kwargs'} not documented
	See Also section not found

File: pandas/core/indexes/datetimes.py [hunk 2]
```diff
@@ -2237,6 +2238,22 @@ def date_range(start=None, end=None, periods=None, freq='D', tz=None,
...
+    >>> pandas.date_range('20181003', periods=2)
```
Comment (TomAugspurger) [inline pandas/core/indexes/datetimes.py]: Can you do these in iso format `2018-10-03`?

**Review by TomAugspurger — APPROVED** [review]

**Comment by TomAugspurger** [comment]
Just a couple small comments. Can ignore the kwargs issue.

File: pandas/core/indexes/datetimes.py (hunk 2 above)
Comment (nipunsadvilkar) [reply pandas/core/indexes/datetimes.py]: will do it

Comment (nipunsadvilkar) [reply pandas/core/indexes/datetimes.py]: Done!

**Comment by nipunsadvilkar** [comment]
@TomAugspurger : Please check changes now. I've made the changes as you suggested

File: pandas/core/indexes/datetimes.py [hunk 3]
```diff
@@ -2237,6 +2238,21 @@ def date_range(start=None, end=None, periods=None, freq='D', tz=None,
     Returns
     -------
     rng : DatetimeIndex
+
+    See Also
+    ________
```
Comment (kynan) [inline pandas/core/indexes/datetimes.py]: Consider using `----` for consistency

Comment (nipunsadvilkar) [reply pandas/core/indexes/datetimes.py]: Ohh silly mistake! thanks for figuring it out

**Comment by nipunsadvilkar** [comment]
Have made all the required changes as per reviewers feedback. Can be merged now!

**Comment by TomAugspurger** [comment]
@nipunbatra I pushed a commit with some whitespace normalization so that the doctests would pass.

**Comment by nipunbatra** [comment]
@TomAugspurger I think you pinged the wrong Nipun :D 
Pinging https://github.com/nipunsadvilkar

**Comment by TomAugspurger** [comment]
😬 sorry!

**Comment by nipunsadvilkar** [comment]
@TomAugspurger : Thanks! Do I need to make any other changes?

**Comment by TomAugspurger** [comment]
Nope, we're good thanks!

**Pull request merged**